src/pages/index.js

```diff
@@ -103,41 +141,123 @@ function closeModalOverlay(event) {
 
 function handleEditFormSubmit(event) {
   event.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal, settings);
+
+  // change text content to "Saving...."
+  const submitButton = event.submitter;
+  // submitButton.textContent = "Saving...";
+  setButtonText(submitButton, true, "Save", "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      // TODO - Use data argument instead of the input values
+      profileName.textContent = editModalNameInput.value;
+      profileDescription.textContent = editModalDescriptionInput.value;
+      closeModal(editModal, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // TODO - Call setButtonText instead
+      submitButton.textContent = "Save";
+    });
 }
 
+// TODO - Implement loading text for all other form submissions
+
 function handleAddCardSubmit(event) {
   event.preventDefault();
   const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  event.target.reset();
-  disabledButton(cardSubmitButton, settings);
-  closeModal(cardModal);
+
+  api
+    .addNewCards(inputValues)
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+      event.target.reset();
+      disabledButton(cardSubmitButton, settings);
+      closeModal(cardModal);
+    })
+    .catch((error) => console.error(error));
+}
```

Comment:
Please, reuse `setButtonText` everywhere if you want to change the button text


```js
    .finally(() => {
      setButtonText(submitButton, false);
    });

```


You don't need to write the **texts** because they are stated inside `setButtonText` **by default**